### PR TITLE
Improve API, add more documentation to readme

### PR DESCRIPTION
**Pull request by YYYasin19**
Simply some instructions for any curious early readers :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,40 @@ A python package for efficient pickling.
...
+```
+
+## Usage
+
+Using `slim-trees` does not affect your training pipeline.
+Simply replace calls to `pickle.dumps` with `slim_trees.dump_sklearn_compressed`:
+
+```python
+# example, you can also use other Tree-based models
+from sklearn.tree import DecisionTreeClassifier
```
Comment (pavelzw) [inline README.md:25]: Let's maybe change this to `RandomForestRegressor`, the savings for a single tree are not worth the effort 😅

File: README.md [hunk 2]
```diff
@@ -9,6 +9,40 @@ A python package for efficient pickling.
...
+```bash
+pip install slim-trees
+# or
+mamba install slim-trees -c conda-forge
+```
+
+## Usage
+
+Using `slim-trees` does not affect your training pipeline.
+Simply replace calls to `pickle.dumps` with `slim_trees.dump_sklearn_compressed`:
```
Comment (pavelzw) [inline README.md:21]:
```suggestion
Simply replace `pickle.dump` calls with `slim_trees.dump_sklearn_compressed`:
```

**Comment by jonashaag** [comment]
While you’re at it you might also want to add more general description for the package. Just copy and paste some stuff from the various presentation materials that we have :) would be nice to also have reduced version of the benchmark table that we get in PRs.

File: README.md (hunk 2 above)
Comment (pavelzw) [reply README.md:21]: Also, `dump_sklearn_compressed` does not have the same signature as `pickle.dump`.
`dump_sklearn_compressed` takes a path as argument while `pickle.dump` takes a file as argument. 
`slim_trees.sklearn_tree.dump_sklearn` would be a drop-in replacement for `pickle.dump`.

Comment (YYYasin19) [reply README.md:21]: Yes, let's include both maybe

**Comment by jonashaag** [comment]
I'm a big fan of adding https://pre-commit.ci btw :)

**Comment by pavelzw** [comment]
Looking good! 

Although we should also say something about LightGBM as well.

Also, for the benchmark image, I suggest adding a dark and white mode image. These can be displayed depending on the current system color scheme:
```md
![benchmark](./.github/assets/benchmark-light.png#gh-light-mode-only)
![benchmark](./.github/assets/benchmark-dark.png#gh-dark-mode-only)
```
But this can also wait for a follow-up PR.

**Review by pavelzw — APPROVED** [review]

**Pull request merged**